[DOCS]  3.2.0 release (#3952)
percona/pmm: documentation/docs/release-notes/3.2.0.md

# Percona Monitoring and Management 3.2.0 

| **Release date** | May 29th, 2025                                                                                  |
| ----------------- | :---------------------------------------------------------------------------------------------- |
| **Installation** | [Installing Percona Monitoring and Management](../quickstart/quickstart.md) |
| **Upgrade**| [Migrate PMM 2 to PMM 3](../pmm-upgrade/migrating_from_pmm_2.md) |

Percona Monitoring and Management (PMM) is an open source database monitoring, management, and observability solution for MySQL, PostgreSQL, and MongoDB. PMM empowers you to: 

- monitor the health and performance of your database systems
- identify patterns and trends in database behavior
- diagnose and resolve issues faster with actionable insights
- manage databases across on-premises, cloud, and hybrid environments

## Release summary

> PMM 3.2.0 comes with better support for Amazon Linux 2023 and MySQL 8.4 replication, an upgrade to Grafana 11.6, secure ClickHouse connections, and a new backup dashboard for MongoDB. 
> 
> QAN gets faster filtering. Plus, many bug fixes and performance improvements!

Reviewer: ```suggestion
PMM 3.2.0 comes with better support for Amazon Linux 2023 and MySQL 8.4 replication, an upgrade to Grafana 11.6, secure ClickHouse connections, a new backup dashboard for MongoDB, faster filtering for QAN.  This is in addition to many bug fixes and performance improvements!
```

To get started, open the [PBM Details dashboard](../reference/dashboards/dashboard-mongodb-PBM-details.md) and see how it works with your MongoDB environments.

![PBM dashboard](../images/PBM_Dashboard.png)

Future releases will improve the **Backup history** panel to better display certain error states and special conditions (like *stuck* or *incompatible* backups) that are currently only visible when using `PBM's pbm status` command directly.

### Improved Query Analytics filter performance
We've optimized QAN filter loading performance to reduce the number of processed rows by up to 95% in large environments. This means that the filters on the **PMM Query Analytics** page page now load faster, which makes the interface more responsive. 

![QAN filters](../images/QAN_filters.png)

### Nomad integration for future extensibility

We've integrated [HashiCorp Nomad](https://developer.hashicorp.com/nomad) into PMM to lay the groundwork for future feature development. This workload orchestrator will initially manage monitoring agents to improve service delivery for Percona customers.

> For advanced users, the Nomad API (internal port `4646`) is accessible via the standard PMM HTTPS port using the `/nomad` prefix and requires Administrator privileges.

Reviewer: I think you mean URL and not prefix (i.e. https://\<pmm-server\>/nomad) but if I'm misinterpreting that then it should be reworked because I'm not sure what else you meant...

> Nomad is disabled by default and its increase in Docker image size has no performance impact when not enabled. For more information, see [Nomad configuration](../reference/nomad.md). 

Reviewer: ```suggestion
Nomad is disabled by default.  Even though it increases the overall Docker image size, it has no performance impact when not enabled. For more information, see [Nomad configuration](../reference/nomad.md). 
```

    You can customize the retention by updating the Time-To-Live (TTL) expression (`event_date + INTERVAL X DAY DELETE`) in the ClickHouse configuration.

- [PMM-13937](https://perconadev.atlassian.net/browse/PMM-13937) - Improved the output of the `pmm-admin list` command to exclude the `nomad_agent` from the output when it's stopped or not running. This helps you get a cleaner representation of active PMM services.

- [PMM-13869](https://perconadev.atlassian.net/browse/PMM-13869) - PMM now outputs Nginx access logs in LOGFMT format, ensuring consistency across all PMM components. This also improves compatibility with log analysis tools like Loki and VictoriaLogs, which means that you can analyze logs directly without reformatting or manual parsing.

- [PMM-13171](https://perconadev.atlassian.net/browse/PMM-13171) - Added support for connecting to external ClickHouse databases with custom user credentials using the `PMM_CLICKHOUSE_USER` and `PMM_CLICKHOUSE_PASSWORD` environment variables. 

> - [PMM-13832](https://perconadev.atlassian.net/browse/PMM-13832) - We've revised the [PMM installation documentation](https://docs.percona.com/percona-monitoring-and-management/3/install-pmm/index.html) to enhance technical accuracy, clarity, and usability. As we continue to refine the topics in this chapter, we welcome your feedback to help us make the documentation even better.

Author: we want to encourage users to give feedback on the install chapter so we can improve it even further with the next release. we generally document doc improvements in relNotes.

> - [PMM-13832](https://perconadev.atlassian.net/browse/PMM-13832) - We've revised the [PMM installation documentation](https://docs.percona.com/percona-monitoring-and-management/3/install-pmm/index.html) to enhance technical accuracy, clarity, and usability. As we continue to refine the topics in this chapter, we welcome your feedback to help us make the documentation even better.

Reviewer 2: oh, ok

### Secure external ClickHouse connections 
PMM now supports connecting to external ClickHouse servers using authenticated credentials. 

You can now set the `PMM_CLICKHOUSE_USER` and `PMM_CLICKHOUSE_PASSWORD` environment variables when deploying PMM Server to enable access to password-protected ClickHouse instances, such as those managed by Altinity’s operator. 

This means that previous authentication errors are resolved, and the less secure practice of embedding credentials in the connection string is no longer necessary.

### Feature Compatibility Version (FCV) panel in MongoDB dashboards
PMM now shows the current MongoDB Feature Compatibility Version (FCV) in the  **MongoDB Replica Set** and **MongoDB Sharded Cluster Summary** dashboards. 

This panel helps you quickly see if your cluster is running the expected FCV—especially after upgrades, when the FCV might not match the MongoDB version yet.

![FCV panels](../images/FCV_panel.png)

> ### Label-based access control for Query Analytics
> QAN now supports label-based access control (LBAC), enabling fine-grained, role-based visibility into query data. This restricts users to viewing only the databases and services assigned to their roles, with filter options automatically adjusted based on their permission labels.
> 
> This makes it easy to limit access by technology (e.g., MySQL-only), environment (e.g., production), or region, enhancing both security and clarity.
> 
> To test it, set up a PMM Server, connect MySQL and PostgreSQL databases, enable access control, and define roles using [LBAC selectors](../admin/roles/access-control/intro.md).

Reviewer 3: ```suggestion
```

We are not shipping it, it goes to 3.3.0.

# Percona Monitoring and Management 3.2.0 

| **Release date** | May 15th, 2025                                                                                  |
| ----------------- | :---------------------------------------------------------------------------------------------- |
| **Installation** | [Installing Percona Monitoring and Management](../quickstart/quickstart.md) |
| **Upgrade**| [Migrate PMM 2 to PMM 3](../pmm-upgrade/migrating_from_pmm_2.md)

Percona Monitoring and Management (PMM) is an open source database monitoring, management, and observability solution for MySQL, PostgreSQL, and MongoDB.

PMM empowers you to: 

- monitor the health and performance of your database systems
- identify patterns and trends in database behavior
- troubleshoot issues effectively
- execute database management tasks seamlessly, whether your databases are on-premises or in the cloud

## Release summary

## Release highlights

## Improvements

- [PMM-13406](https://perconadev.atlassian.net/browse/PMM-13869)- We've integrated HashiCorp Nomad into PMM — a powerful workload orchestrator designed to improve resource allocation and enable future extensibility.

> Nomad is disabled by default, and although the Docker image size has increased slightly, this additional footprint has no impact on performance when Nomad remains disabled.

Reviewer 3: ```suggestion
Nomad is disabled by default, and although the Docker image size has slightly increased, this additional footprint has no impact on performance when Nomad remains disabled.
```
